design-docs/006-tag-delete.md

```diff
@@ -0,0 +1,91 @@
+# Deleting Tags
+
+We like our tags immutable. That gives Icechunk users a lot of power.
+For example, they can always cache anything coming out
+from an Icechunk session initiated on a tag:
+```python
+repo = icechunk.Repository.open(...)
+session = repo.readonly_session(tag="v42")
+store = session.store
+```
+
+All data read from `store` will be immutable, it can be cached for as
+long as needed. This is only true if tags cannot be edited. If
+somebody can edit `v42` and move it to a different snapshot,
+there are no longer any guarantees.
+
+## The problems
+
+* If a user makes a spelling mistake, or they assign a tag to the wrong
+snapshot, they cannot fix it. This is a minor problem, and we made
+an explicit trade-off here. The benefits justify this minor inconvenience.
+The will have to create a new, different tag.
+* In a world in which we want to expire on squash history, tags will have to
```

Comment:
Could all history except for tagged snapshots be expired?

Reply:
It could. One of the trickiest things in this topic, in my experience, is to decide how to define the multiple ways in which a user could want to expire. By commit age, by branches, by tags, and everything in between.

Initially I'm thinking something like "expire all commits older than X". If there are tags pointing to expirable snapshots, we would give them the option to fail or delete them.